libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1837,12 +1838,15 @@ GCS_MAVLINK::update_receive(uint32_t max_time_us)
         bool parsed_packet = false;
 
         // Try to get a new message
-        if (mavlink_frame_char_buffer(channel_buffer(), channel_status(), c, &msg, &status) == MAVLINK_FRAMING_OK) {
+        const auto framing_status = mavlink_frame_char_buffer(channel_buffer(), channel_status(), c, &msg, &status);
+        if (framing_status != MAVLINK_FRAMING_INCOMPLETE) {
             hal.util->persistent_data.last_mavlink_msgid = msg.msgid;
-            packetReceived(status, msg);
+            packetReceived(framing_status, status, msg);
```

Comment:
need to be careful not to propagate bad CRC pkts up to higher levels (vehicle code)